.github/workflows/lint.yml

```diff
@@ -61,15 +61,54 @@ jobs:
           # Warnings in header files are unavoidable, so we filter out submodule headers after analysis
           # Might be wrong to assume cppcheck always reports issues using 3 lines
            awk -v nlines=2 '/submods\/amrex/ || /submods\/googletest/ {for (i=0; i<nlines; i++) {getline}; next} 1' \
-            < cppcheck.txt > amr-wind-static-analysis.txt
+            < cppcheck.txt > cppcheck-warnings.txt
       - name: Report
-        working-directory: build-lint
+        working-directory: build-cppcheck
         run: |
           echo "::add-matcher::.github/problem-matchers/gcc.json"
-          # Might be wrong to assume cppcheck always reports issues using 3 lines
-          WARNINGS_TMP=$(wc -l < amr-wind-static-analysis.txt | xargs echo -n)
+          # Crude warning number approximation
+          WARNINGS_TMP=$(wc -l < cppcheck-warnings.txt | xargs echo -n)
+          WARNINGS=$(bc <<< "$WARNINGS_TMP/3")
+          printf "Approximately %s warnings\n" "${WARNINGS}" >> cppcheck-warnings.txt
+          cat cppcheck-warnings.txt
+  clang-tidy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{github.token}}
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Dependencies
+        run: |
+          sudo apt-get install -y clang-tidy-9
+          sudo update-alternatives --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-9 100
+      - name: Configure
+        run: |
+          echo "NPROCS=$(nproc)" >> $GITHUB_ENV
+          cmake \
+          -Bbuild-clang-tidy \
+          -DCMAKE_CXX_COMPILER:STRING=clang++ \
+          -DCMAKE_C_COMPILER:STRING=clang \
+          -DAMR_WIND_ENABLE_MPI:BOOL=OFF \
+          -DAMR_WIND_ENABLE_TESTS:BOOL=ON \
+          -DAMR_WIND_TEST_WITH_FCOMPARE:BOOL=OFF \
+          -DAMR_WIND_ENABLE_MASA:BOOL=OFF \
+          -DAMR_WIND_ENABLE_ALL_WARNINGS:BOOL=OFF \
+          -DAMR_WIND_ENABLE_CLANG_TIDY:BOOL=ON \
+          -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON \
+          ${{github.workspace}}
+      - name: Check
+        working-directory: build-clang-tidy
+        run: cmake --build . -- -j ${{env.NPROCS}} 2> clang-tidy-warnings.txt
+      - name: Report
+        working-directory: build-clang-tidy
+        run: |
+          echo "::add-matcher::.github/problem-matchers/gcc.json"
+          # Crude warning number approximation
+          WARNINGS_TMP=$(wc -l < clang-tidy-warnings.txt | xargs echo -n)
           WARNINGS=$(bc <<< "$WARNINGS_TMP/3")
-          printf "%s warnings\n" "${WARNINGS}" >> amr-wind-static-analysis.txt
-          cat amr-wind-static-analysis.txt
-          # Just report for now and don't fail
-          #exit ${WARNINGS}
+          printf "Approximately %s warnings\n" "${WARNINGS}" >> clang-tidy-warnings.txt
```

Comment:
Looking at the outputs in the Checks tab, it looks like `clang-tidy` outputs a lot more than 3 lines per warning. It also shows the code path it took to arrive at that warning.

Reply:
Yeah, I could try to parse for `warning:`. I mostly just wanted something to show as 0 warnings in the report once we get there, but I can try to make something actually count the warnings.